Updating the role to 2019

santiagomr opened the pull request:
Hi, thanks for sharing the role. In our organization we currently use Jitsi Meet. We set up our own instance based on your role, but modifications were necessary to make it functional to the current stable version of Jitsi Meet. We also fixed issues reported by users.

Here are the modifications, they include changes in:
- Debconf options
- Nginx configuration files
- Jitsi configuration files
- Use of personalized certificates
- Order of execution of tasks

santiagomr committed:
Updating role to 2019

santiagomr committed:
Updating role to 2019 keeping the original code style

santiagomr committed:
Updating README.md

santiagomr committed:
Removing comments

santiagomr committed:
Customization of app parameters and organization logos

AmarOk1412 commented:
@santiagomr I used your role to deploy a Jitsi instance. I only got one problem so far, jitsi-videobridge service doesn't start correctly due to https://community.jitsi.org/t/error-when-starting-jvb/16845

I had to remove `$JVB_EXTRA_JVM_PARAMS` from `/etc/jitsi/videobridge/config`

Also the final nginx start doesn't seems to work, I had to restart it manually :)

santiagomr committed:
Removed the deprecated parameter JVB_EXTRA_JVM_PARAMS

santiagomr commented:
Thanks for your feedback @AmarOk1412. As you mentioned, the parameter `$JVB_EXTRA_JVM_PARAMS` was deprecated and is no longer part of the `/etc/jitsi/videobridge/config` file. Including this parameter in the file does not let the service start. To correct it, I updated the [template](https://github.com/UdelaRInterior/ansible-role-jitsi-meet/commit/a3e2de5e18ec12906ab4d321dbf527636da21eb7#diff-f48522d054b93cca09db1f9c7b6f3b44).

Regarding the final nginx start that doesn't work (to which the services prosody, jicofo and jitsi-videobridge are added), it seems to be due to the handlers are skipping, although I still can't determine why this happens.

```bash
RUNNING HANDLER [jitsi_meet : restart nginx] ************************
skipping: [my-server.org]
RUNNING HANDLER [jitsi_meet : restart prosody] ************************
skipping: [my-server.org]
RUNNING HANDLER [jitsi_meet : restart jicofo] ************************
skipping: [my-server.org]
RUNNING HANDLER [jitsi_meet : restart jitsi-videobridge] ************************
skipping: [my-server.org]
```
do you have any idea?

AmarOk1412 commented:
Oh you did the fix even before I send a patch :D

For the nginx, I still don't know, I generally work on hosting things each Thursday. Last time I checked it seems that during the final `nginx` restart, the port 443 was used by something (but I didn't check further.

santiagomr commented:
I will try then to check more in depth. Apparently we both have the same problem caused by different reasons.

AmarOk1412 commented:
Possibly, or just the same reason, I did a lot of things since the day I applied your roles. I probably just forgot the initial problem.

santiagomr committed:
Merge branch 'master' into ui-customization

santiagomr committed:
Parameterization of the images path

AmarOk1412 commented:
Ok. It is Thursday, I cleaned my VM and retried with the updated role.

This is the error for nginx:

```
~# systemctl status nginx.service 
● nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Thu 2019-05-23 18:53:08 EDT; 1min 32s ago
     Docs: man:nginx(8)
  Process: 19525 ExecStart=/usr/sbin/nginx -g daemon on; master_process on; (code=exited, status=1/FAILURE)
  Process: 19522 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
 Main PID: 10285 (code=exited, status=0/SUCCESS)

May 23 18:53:05 garfield nginx[19525]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
May 23 18:53:06 garfield nginx[19525]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
May 23 18:53:06 garfield nginx[19525]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
May 23 18:53:07 garfield nginx[19525]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
May 23 18:53:07 garfield nginx[19525]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
May 23 18:53:08 garfield nginx[19525]: nginx: [emerg] still could not bind()
May 23 18:53:08 garfield systemd[1]: nginx.service: Control process exited, code=exited status=1
May 23 18:53:08 garfield systemd[1]: Failed to start A high performance web server and a reverse proxy server.
May 23 18:53:08 garfield systemd[1]: nginx.service: Unit entered failed state.
May 23 18:53:08 garfield systemd[1]: nginx.service: Failed with result 'exit-code'.
```

and indeed:

```
~# netstat -pluton | grep 443
tcp6       0      0 10.0.0.77:443           :::*                    LISTEN      15676/java           off (0.00/0/0)
tcp6       0      0 54.39.239.121:443       :::*                    LISTEN      15676/java           off (0.00/0/0)
```

The guilty process:

```
~# systemctl status jitsi-videobridge.service 
● jitsi-videobridge.service - Jitsi Videobridge
   Loaded: loaded (/lib/systemd/system/jitsi-videobridge.service; enabled; vendor preset: enabled)
   Active: active (running) since Thu 2019-05-23 18:52:26 EDT; 5min ago
 Main PID: 15676 (java)
```

AmarOk1412 commented:
@santiagomr IMO, two solutions:

1. Change the handler to stop all services then restart.
2. Add a task to stop all services and then the handler will do its job. I think it's the best solution.

I did it on my instance and it works perfectly.

What do you think? Do you want a PR on your repo?

santiagomr commented:
@AmarOk1412 the problem I'm having is actually that the handlers (as they are defined now) are skipped even though they have been triggered correctly.

I gladly accept your PR if you already have it resolved.

AmarOk1412 commented:
Sure. I will send you the PR this week

Commit:
handlers: fix nginx restart

Because jitsi-video-bridge can take the port 443, nginx should
ensure that all jitsi services are correctly stopped.

santiagomr committed:
Merge pull request #1 from AmarOk1412/fix/nginx

handlers: fix nginx restart

santiagomr committed:
Name of tasks in English - Trying customize the welcome page from language files

santiagomr committed:
Improvement in idempotency when customizing the welcome page from language file

santiagomr committed:
ui-customization disabled by default - Unnecessary quotes in interface_config.js.j2

santiagomr committed:
Merge branch 'ui-customization'

santiagomr committed:
Special case at customizing ui when English language is chosen

santiagomr committed:
Update meta info

santiagomr committed:
Update README.md

santiagomr committed:
Updating meta to publish the role in Ansible Galaxy

santiagomr committed:
Improving syntax score

AmarOk1412 commented:
Hei @santiagomr 

Also, note. I add a task on my ansible roles to run `update-ca-certificates` for Jicofo (https://github.com/jitsi/jicofo/blob/master/README.md#certificates)

santiagomr committed:
Set the pipefail option when the shell module is used

msheiny commented:
hey folks! apologies for the delay in response here. We are no longer maintaining this repository but you are obviously welcome to fork as needed. Happy video chatting!

santiagomr committed:
Specifying role_name

santiagomr committed:
Update .gitignore

santiagomr committed:
Release of version 2.0.0

santiagomr committed:
Improvement: added a variable type list to manage STUN servers

Dkmarce committed:
Merge pull request #6 from UdelaRInterior/version-2.0.0

Release of version 2.0.0

fabiogermann committed:
Update jitsi_meet_config.js.j2

fabiogermann committed:
Update main.yml

fabiogermann committed:
Update main.yml

Commit:
changes necessary to make it work behind a hardware nat firewall

santiagomr committed:
Settings to expose /colibri/stats over HTTPS

tabacha committed:
Fix https://github.com/UdelaRInterior/ansible-role-jitsi-meet/issues/9

tabacha committed:
lower case of True/Fale to true/false if it input is boolean

tabacha committed:
Make ufw firewall ports configurable

tabacha committed:
Make ufw firewall ports configurable

tabacha committed:
Enable auth

tabacha committed:
the jigasi sip-communicator.properties file has nothing to do with "sip calls"

santiagomr committed:
Merge pull request #11 from ADFC-Hamburg/ufw-firewports-configurable

Ufw firewports configurable

tabacha committed:
Remove double jitsi_meet_disable_third_party_requests and change default from true to false, see #10

santiagomr committed:
Merge pull request #10 from ADFC-Hamburg/disable_third_party_requests

Disable third party requests

santiagomr committed:
Merge remote-tracking branch 'tabacha/enable-auth'

santiagomr committed:
Preparing v3.0.0

santiagomr committed:
Merge branch 'colibri-stats'

santiagomr committed:
Integrating colibri statistics

santiagomr committed:
Merging #7 - Run behind NAT

santiagomr committed:
Release v3.0.0

tabacha committed:
Fix diffrent statistics settings in videobridge_sip-communicator.properties.j2 and videobridge_config.js

santiagomr committed:
Merge pull request #13 from ADFC-Hamburg/statistics-in-sip-config

Fix diffrent statistics settings in videobridge_sip-communicator.prop…

santiagomr committed:
Release v3.0.1

tabacha committed:
Cfg cleanup and make Performance optimizations possible

tabacha committed:
Rename Cfg: jitsi_meet_disable_third_party_requests -> jitsi_meet_enable_third_party_requests

tabacha committed:
Rename Cfg: jitsi_meet_desktop_sharing_.._disabled -> jitsi_meet_enable_desktop_sharing_..

tabacha committed:
Rename Cfg: jitsi_meet_disable_video_background -> jitsi_meet_show_video_background

tabacha committed:
Rename Cfg: jitsi_meet_disable_audio_levels -> jitsi_meet_show_audio_levels

santiagomr committed:
Merge remote-tracking branch 'tabacha/performance_optimize_posible'

santiagomr committed:
Some fixes after merge #14

santiagomr committed:
Manual merge from ADFC-Hamburg:rename_disable_variables

santiagomr committed:
Add CI workflow in Actions

santiagomr committed:
fixing error 701 in ansible-lint

santiagomr committed:
Fixes #16 and several updates in videobridge configuration

santiagomr committed:
Merge branch 'master' of github.com:UdelaRInterior/ansible-role-jitsi-meet

santiagomr committed:
Release 4.0.0

tabacha committed:
Make all templates configurable

tabacha committed:
Make all templates configurable: fix name to jitsi_meet_nginx_config_template

santiagomr committed:
Merge pull request #22 from ADFC-Hamburg/dynamic_template

All jinja templates from parameterized paths

santiagomr committed:
Release v4.1.0

Commit:
gpg apt key chamged
